waterlog/src/utils.js

```diff
@@ -1,186 +1,197 @@
+    let sensorInColor = defaultIcon,
+      sensorOutColor = defaultIcon,
+      segmentColor = defaultColor;
+    if (sensorIn.status.toLowerCase() === 'faulty') {
+      sensorInColor = sensorFaultIcon;
+    }
+    if (sensorOut.status.toLowerCase() === 'faulty') {
+      sensorOutColor = sensorFaultIcon;
+    }
+    if (segment.status.toLowerCase() === 'leak') {
+      segmentColor = options.colors.errorColor;
+    }
+
+    return (
+      <div>
+        <Polyline
+          positions={[[sensorIn.lat, sensorIn.lon], [sensorOut.lat, sensorOut.lon]]}
+          color={segmentColor}
+          weight={options.lineWeight}
+        >
+          <Popup>
+            <span>{'segment ' + segment.id + '\n status ' + segment.status}</span>
+          </Popup>
+        </Polyline>
+        <Marker position={[sensorIn.lat, sensorIn.lon]} opacity={1} key={sensorIn.id} icon={sensorInColor}>
+          <Popup>
+            <span>{'sensor ' + sensorIn.id + '\n status ' + sensorIn.status}</span>
+          </Popup>
+        </Marker>
+        <Marker position={[sensorOut.lat, sensorOut.lon]} opacity={1} key={sensorOut.id} icon={sensorOutColor}>
+          <Popup>
+            <span>{'sensor ' + sensorOut.id + '\n status ' + sensorOut.status}</span>
+          </Popup>
+        </Marker>
+      </div>
+    );
+  });
 }
 
 export function levelToIntensity(level, maxIntensity = 5) {
-	if (!level) {
-		return 0;
-	}
-	switch (level.toLowerCase()) {
-		case 'high':
-			return maxIntensity;
-		case 'medium':
-			return Math.ceil(maxIntensity / 2);
-		case 'low':
-			return 1;
-		default:
-			return 0;
-	}
+  if (!level) {
+    return 0;
+  }
+  switch (level.toLowerCase()) {
+    case 'high':
+      return maxIntensity;
+    case 'medium':
+      return Math.ceil(maxIntensity / 2);
+    case 'low':
+      return 1;
+    default:
+      return 0;
+  }
+}
+
+export function getHeatMapData({ monitorsCoordinates, segmentCoordinates }) {
+  let monitorMapData = monitorsCoordinates.map(mon => {
+    return [mon.lat, mon.long, levelToIntensity(mon.faultLevel, mapOptions.maxIntensity)]
+  });
+  let segmentMapData = segmentCoordinates.map(seg => {
+    return [seg.lat, seg.long, levelToIntensity(seg.faultLevel, mapOptions.maxIntensity)]
+  });
+
+  let heatMapData = monitorMapData.concat(segmentMapData);
+  return heatMapData;
 }
 
-const southWest = [ -25.944586, 28.189546 ];
-const northEast = [ -25.661871, 28.451147 ];
+const southWest = [-25.944586, 28.189546];
+const northEast = [-25.661871, 28.451147];
 const heatBackgroundConst = 0.91;
 
 export const mapOptions = {
-	southWest: southWest,
-	northEast: northEast,
-	maxIntensity: 5,
-	centerPosition: [ -25.783425, 28.336046 ],
-	defaultZoom: 17,
-	heatBackgroundConst: heatBackgroundConst,
-	rectangleBounds: [ southWest.map((e) => e / heatBackgroundConst), northEast.map((e) => e * heatBackgroundConst) ]
+  southWest: southWest,
+  northEast: northEast,
+  maxIntensity: 5,
+  centerPosition: [-25.783425, 28.336046],
```

Comment:
mimic what was done with `southWest` and `northEast `

Reply:
done